windowsdoc.md

```diff
@@ -0,0 +1,67 @@
+* Docker
+
+### ***Best Practises***:  
+You should install nodejs on your system if your Workflow contains [Javascript Expressions](http://www.commonwl.org/v1.0/CommandLineTool.html#InlineJavascriptRequirement).
+
+### ***Installation***:  
+**Using Pip**:  
+You can install cwltool on windows using pip  
+`pip install cwltool`
+
+**Developing from source**:  
+`git clone https://github.com/common-workflow-language/cwltool.git`  
+`cd cwltool`  
+`python setup.py develop`
+
+***Note:*** In order to test if cwltool has been successfully installed on your windows system, run `cwltool` on your `cmd`. A screen showing help instructions should be there.
+
+### ***Running Cwltool's Unittest***  
+If you want to run cwltool's unit test, go to cwltool's repository on your system and run  
+`python setup.py test`
+
+### ***Running Conformance tests using cwltool***  
+Download cwltest using `pip install cwltest`  
+Clone `Common-Workflow-language` repository using `git clone https://github.com/common-workflow-language/common-workflow-language.git`   
+change directory using `cd common-workflow-language/v1.0`  
+Run Tests with `cwltest --test conformance_test_v1.0.yaml -j 4 --tool cwltool`, here `-j` arg is used to run multiple tests in parallel.
+
+&nbsp;
+&nbsp;
+
+> ***Installing cwltool on windows is smooth but you may face some problems in setting up Docker. Some of the common issues and their solution are mentioned below.***
+
+**Docker doesn't work on windows, even outside cwltool.**  
+Make sure you followed all instructions carefully while installing Docker. Especially setting up Environment variables. Consult online Docker community if problem persists.
+
+**You Local Drives are not being Shared with Docker Container.**  
+* ***On native Docker for windows(supported by Windows 10):***  
+On your tray, next to your clock, right-click on Docker, then click on Settings, there you'll find the Shared Drives: Here you can share your Drives with Docker.  
+If you encounter a Firewall issue, you can [refer this post](https://blog.olandese.nl/2017/05/03/solve-docker-for-windows-error-a-firewall-is-blocking-file-sharing-between-windows-and-the-containers/).
+
+* ***On Docker Toolbox:***  
+Docker Toolbox uses Virtualbox to create linux base on which Docker machine runs. Your Docker Container will be created inside Virtualbox. To share you Drives
+in virtualbox, go to ****Virtualbox->settings->shared folders->Machine Folders****
+Here Map your Drives you want to share with your Docker Container.  
+If you want to keep these settings permanent(Recommended!), You should mark `make permanent` checkbox or else these settings will be erased everytime your virtualbox closes.
+
+**In a Docker Container with Shared Drives, few Files are shown on `ls` but most of them doesn't.**  
+This means your Drives/Folders are not shared properly. When a path is mentioned in docker, it gets cached. So the few folder names that appear on `ls` is because of that.  
```

Comment:
- When a path is mentioned in docker, it gets cached. So the few folder names that appear on `ls` is because of that.   -> Docker uses caching, which may result in not all files and folders being listed on `ls`.